fpsd/classify.py

```diff
@@ -0,0 +1,277 @@
+            cv = cross_validation.StratifiedKFold(y_val, n_folds=self.k,
+                                                  shuffle=True)
+        elif self.world_type == "open":
+            pass  # TODO
+
+        fpr_arr, tpr_arr, metrics_all_folds = [], [], []
+        for i, (train, test) in enumerate(cv):
+
+            fold_timestamp = datetime.datetime.now().isoformat()
+            y_train, y_test = y_val[train], y_val[test]
+
+            if self.feature_scaling:
+                scaler = preprocessing.StandardScaler().fit(x_val[train])
+                x_train = scaler.transform(x_val[train])
+                x_test = scaler.transform(x_val[test])
+            else:
+                x_train, x_test = x_val[train], x_val[test]
+
+            trained_model = self.train_single_fold(x_train, y_train)
+            pred_probs = self.score(x_test, trained_model)
+
+            filename_kfold = '{}_{}_undefended_frontpage_{}_model_{}_fold_{}_world.pkl'.format(
+                fold_timestamp, self.model_timestamp, self.model_type,  i, self.world_type)
+            fold_to_save = {'trained_object': trained_model,
+                            'y_true': y_test, 'y_predicted': pred_probs}
+            self.pickle_results(filename_kfold, fold_to_save)
+
+            # Metrics computation
+            # Compute ROC curve and area under the ROC curve
+            eval_metrics = evaluation.get_metrics(y_test, pred_probs)
+            metrics_all_folds.append(eval_metrics)
+            fpr_arr.append(eval_metrics['fpr'])
+            tpr_arr.append(eval_metrics['tpr'])
+
+            # Save results of metrics in database
+            self.db.save_fold_of_model(eval_metrics, self.model_timestamp, fold_timestamp)
+
+        auc = evaluation.plot_allkfolds_ROC(self.model_timestamp, cv,
+                                            fpr_arr, tpr_arr)
+
+        print("Classifier {} trained! AUC: {}".format(self.model_timestamp,
+                                                      auc))
+
+        avg_metrics = evaluation.get_average_metrics(metrics_all_folds)
+        # Save results of experiment (model evaluation averaged over all
+        # folds) into the database
+        self.db.save_full_model(avg_metrics, self.model_timestamp, self.get_dict())
+
+    def pickle_results(self, pkl_file, to_save):
+        with open(pkl_file, 'wb') as f:
+            pickle.dump(to_save, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+    def _get_model_object(self, model, parameters, n_cores):
+        """This method takes the requested model type and
+        hyperparameters and produces the relevant classifier object.
+
+        Returns:
+            object with the fit() and predict_proba() methods
+            implemented on it
+        """
+
+        if self.model_type == "RandomForest":
+            return ensemble.RandomForestClassifier(
+                n_estimators=self.hyperparameters['n_estimators'],
```

Comment:
Here is my concern: I think this is already difficult to understand for people not very familiar with scikit-learn classifiers and their hyperparameters. I'm worried that by using unpacking here this will become even more difficult to understand - it reduces the number of lines of code, but also negatively impacts readability. Thoughts?